public/pages/HistoricalDetectorDetail/components/HistoricalDetectorControls/HistoricalDetectorControls.tsx

```diff
@@ -0,0 +1,117 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiButton,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+} from '@elastic/eui';
+import { Detector } from '../../../../models/interfaces';
+import { Listener } from '../../../../utils/utils';
+import { DETECTOR_STATE } from '../../../../../server/utils/constants';
+import { HISTORICAL_DETECTOR_ACTION } from '../../utils/constants';
+
+interface HistoricalDetectorControlsProps {
+  detector: Detector;
+  isStoppingDetector: boolean;
+  onEditDetector(): void;
+  onStartDetector(): void;
+  onStopDetector: (
+    action: HISTORICAL_DETECTOR_ACTION,
+    listener?: Listener
+  ) => void;
+  onDeleteDetector(): void;
+}
+export const HistoricalDetectorControls = (
+  props: HistoricalDetectorControlsProps
+) => {
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
+      <EuiFlexItem grow={false} style={{ marginRight: '16px' }}>
+        <EuiPopover
+          id="actionsPopover"
+          button={
+            <EuiButton
+              iconType="arrowDown"
+              iconSide="right"
+              data-test-subj="actionsButton"
+              onClick={() => setIsOpen(!isOpen)}
+              isDisabled={props.isStoppingDetector}
+            >
+              Actions
+            </EuiButton>
+          }
+          panelPaddingSize="none"
+          anchorPosition="downLeft"
+          isOpen={isOpen}
+          closePopover={() => setIsOpen(false)}
+        >
+          <EuiContextMenuPanel>
+            <EuiContextMenuItem
+              key="editDetector"
+              data-test-subj="editDetectorItem"
+              onClick={props.onEditDetector}
+            >
+              Edit historical detector
+            </EuiContextMenuItem>
+            <EuiContextMenuItem
+              key="deleteDetector"
+              data-test-subj="deleteDetectorItem"
+              onClick={props.onDeleteDetector}
+            >
+              Delete historical detector
+            </EuiContextMenuItem>
+          </EuiContextMenuPanel>
+        </EuiPopover>
+      </EuiFlexItem>
+      <EuiFlexItem grow={false} style={{ marginLeft: '0px' }}>
+        {props.isStoppingDetector ? (
+          <EuiButton
+            data-test-subj="isStoppingDetectorButton"
+            onClick={() => {}}
+            isLoading={true}
```

Comment:
Is it possible to just reuse the original <EuiButton> just set `isLoading={props.isStoppingDetector}` ? I think the button will not be clickable if `isLoading=true`, so no need to worry the `onClick` function.

Reply:
That's a good point. Will change to just have the `startDetectorButton` and `stopDetectorButton`.